pkg/client/patch.go

```diff
@@ -59,8 +63,29 @@ func ConstantPatch(patchType types.PatchType, data []byte) Patch {
 	return RawPatch(patchType, data)
 }
 
+type MergeFromWithOptimisticLock struct{}
```

Comment:
I had to read the code to understand what you meant by `OptimisticLock`. This definitely needs godocs.

I also don't understand how that differs from `Update` except that you put less data on the wire and trade it for a local computation of the patch.

Reply:
+1 I'll add a godoc